src/AmalgamatorSession.ts

```diff
@@ -637,12 +637,19 @@ export class AmalgamatorSession extends LoggingDebugSession {
         response: DebugProtocol.Response,
         args: any
     ): Promise<void> {
-        // XXX: Which childDap to send a customRequest to? Probably needs domain specific knowledge.
-        const childResponse = await this.childDaps[0].customRequest(
-            command,
-            args
-        );
-        response.body = childResponse.body;
-        this.sendResponse(response);
+        if (command === 'cdt-gdb-adapter/Memory') {
+            if (args.address === '') {
+                response.body = { data: '', address: '' };
+                response.body.child = this.childDapNames;
+            } else {
+                const childResponse = await this.childDaps[
+                    args.child
```

Comment:
The type here of `args.child` looks like it may be a string based on what I see in cdt-gdb-vscode. Avoid implicit type conversion, make sure to index arrays with numbers.

Reply:
I have updated it

Reply:
You stopped using args.child, in fact args.child seems to be fully unused here now. Please correct that.

Reply:
I have corrected it